api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java

```diff
@@ -350,6 +350,38 @@ public void getDateFormat_shouldReturnAPatternWithFourYCharactersInIt() {
 		Assert.assertEquals("tt.MM.uuuu", OpenmrsUtil.getDateFormat(Locale.GERMAN).toLocalizedPattern());
 		Assert.assertEquals("dd-MM-yyyy", OpenmrsUtil.getDateFormat(new Locale("pt", "pt")).toLocalizedPattern());
 	}
+	@Test
+	public void getDateFormat_shouldReturnTrueWhenDateIsNotNull() {
+		Date date = null;
+		try {
+			SimpleDateFormat sdf= new SimpleDateFormat(MM/dd/yyyy);
+			  date=sdf.parse(value);
+			  if(!value.equals(sdf.format(date))) {
+				  date=null;
+			  }
+			  
+		}catch(ParseException ex) {
+			ex.printStackTrace();
+		}
+		if(!date==null) {
```

Comment:
why place asertions in if statements ? do you intent to run some assertions and not all based on the a given condition ??

Reply:
Does placing assertions in if statements stops the method to run as it was, i thought inserting if statements doesnot change the class and its methods